metrics/prometheus/prometheus.go

```diff
@@ -0,0 +1,320 @@
+		Registry: registry,
+		Puts: &PrometheusRequestStatusMetric{
+			Duration: newHistogram(cfg, registry,
+				"puts_request_duration",
+				"Duration in seconds Prebid Cache takes to process put requests.",
+				timeBuckets,
+			),
+			RequestStatus: newCounterVecWithLabels(cfg, registry,
+				"puts_request",
+				"Count of total requests to Prebid Server labeled by status.",
+				[]string{StatusKey},
+			),
+		},
+		Gets: &PrometheusRequestStatusMetric{
+			Duration: newHistogram(cfg, registry,
+				"gets_request_duration",
+				"Duration in seconds Prebid Cache takes to process get requests.",
+				timeBuckets,
+			),
+			RequestStatus: newCounterVecWithLabels(cfg, registry,
+				"gets_request",
+				"Count of total get requests to Prebid Server labeled by status.",
+				[]string{StatusKey},
+			),
+		},
+		PutsBackend: &PrometheusRequestStatusMetricByFormat{
+			Duration: newHistogram(cfg, registry,
+				"puts_backend_duration",
+				"Duration in seconds Prebid Cache takes to process backend put requests.",
+				timeBuckets,
+			),
+			PutBackendRequests: newCounterVecWithLabels(cfg, registry,
+				"puts_backend",
+				"Count of total requests to Prebid Cache labeled by format, status and whether or not it comes with TTL",
+				[]string{FormatKey},
+			),
+			RequestLength: newHistogram(cfg, registry,
+				"puts_backend_request_size_bytes",
+				"Size in bytes of a backend put request.",
+				requestSizeBuckets,
+			),
+		},
+		GetsBackend: &PrometheusRequestStatusMetric{
+			Duration: newHistogram(cfg, registry,
+				"gets_backend_duration",
+				"Duration in seconds Prebid Cache takes to process backend get requests.",
+				timeBuckets,
+			),
+			RequestStatus: newCounterVecWithLabels(cfg, registry,
+				"gets_backend",
+				"Count of total backend get requests to Prebid Server labeled by status.",
+				[]string{StatusKey},
+			),
+		},
+		Connections: &PrometheusConnectionMetrics{
+			ConnectionsClosed: newSingleCounter(cfg, registry, ConnOpenedKey, "Count the number of open connections"),
+			ConnectionsOpened: newSingleCounter(cfg, registry, ConnClosedKey, "Count the number of closed connections"),
+			ConnectionsErrors: newCounterVecWithLabels(cfg, registry,
+				ConnErrorKey,
+				"Count the number of connection accept errors or connection close errors",
+				[]string{ConnErrorKey},
+			),
+		},
+		ExtraTTL: &PrometheusExtraTTLMetrics{
+			ExtraTTLSeconds: newHistogram(cfg, registry,
+				"extra_ttl_seconds",
+				"Extra time to live in seconds specified",
+				timeBuckets,
+			),
+		},
+		MetricsName: MetricsPrometheus,
+	}
+	promMetrics.ExtraTTL.ExtraTTLSeconds.Observe(5000.00)
+
+	// Should be the equivalent of the following influx collectors
+	// go metrics.CaptureRuntimeMemStats(m.Registry, flushTime)
+	// go metrics.CaptureDebugGCStats(m.Registry, flushTime)
+	promMetrics.Registry.MustRegister(
+		prometheus.NewProcessCollector(prometheus.ProcessCollectorOpts{Namespace: cfg.Namespace}),
+		prometheus.NewGoCollector(),
```

Comment:
The NewGoCollector does not allow you to provide a namespace. Let's chat about this one in person.

Reply:
We discussed to leave out the GoCollector and follow-up in a future PR since it doesn't allow namespace or subsystem config.